boot/boot.js

```diff
@@ -576,10 +579,27 @@ $tw.utils.registerFileType = function(type,encoding,extension,options) {
 	$tw.config.contentTypeInfo[type] = {encoding: encoding, extension: extension, flags: options.flags || [], deserializerType: options.deserializerType || type};
 };
 
-/*
-Given an extension, always access the $tw.config.fileExtensionInfo
-using a lowercase extension only.
-*/
+/**
+ * @typedef ContentTypeInfo
+ * @type {object}
+ * @property {string} encoding
+ * @property {string} extension
+ * @property {Array=} flags
+ * @property {string} deserializerType
+ */
+
+/**
+ * @typedef FileExtensionInfo
+ * @type {object}
+ * @property {string} type
+ */
+
+/**
+ * Given an extension, always access the $tw.config.fileExtensionInfo
+ * using a lowercase extension only.
+ * @param {string} ext - Extension to find a type for
+ * @returns {FileExtensionInfo | null | undefined}
+ */
```

Comment:
Also, commenting on my own code there, but how do you like this style of comments ?

They're [JSDoc](https://jsdoc.app/) type information. It's quite nifty, though not all its features are available in all editors (as they tend to implement TypeScript type hints, which are only a subset of what JSDoc can express).

I don't plan on adding them on _everything_, but I'm adding them as I go and find them useful.